### DISCUSSION TITLE
Question about atomic energy scaling and shifting

### DISCUSSION THREAD

**Discussion by baham2** [deepmodeling/deepmd-kit#4573] [General]
Hello,

I was wondering if the DeePMD package has any support or features for scaling and shifting atomic energies before computing the total energy, specifically through learnable scaling and shifting parameters. I’ve been working on modeling both atomic forces and system energy. While I’ve found that both train and test force errors are centered around zero with a mean absolute error (MAE) of 0.04 eV/Å, which aligns with the literature, the energy error peaks at a non-zero value (0.05 eV). Adjusting the energy and force prefactors hasn't improved this energy error. I’ve verified the quality of my dataset in all aspects (SCF convergence and proper energy and force convergence), and everything appears to be fine. In a separate effort, I trained this system using the Nequip model, where both energy and force errors peaked at zero with reasonable MAEs. This led me to wonder if there could be a similar learnable atomic scaling and shifting mechanism that might improve energy predictions.

Thanks!

**Comment by wanghan-iapcm** [comment]
The DeePMD-kit does the energy shifting. It fits the energy shift value for each element type from the training data, and apply the shift on top of the model prediction. 

In your case, it is recommended a two-step training strategy:
1. a standard training with default parameters. 
2. starting from the model trained by step1, use the --init-model option to load the model parameters and train the model again with larger energy prefactors, smaller force prefactors and a smaller start learning rate, e.g. 1e-4. 
It usually significantly improve the energy accuracy without loss of force and virial accuracy. 

If extreme accuracy is required, you may want to use the [dpa-2 model](https://www.nature.com/articles/s41524-024-01493-2) , which usually shows comparable or even better accuracy than nequip.

**Comment by baham2** [comment]
Thanks for the reply. I followed your instruction. In the second step I used the pretrained model (--init-model 'checkpoint'). Also, I set energy start and stop prefactors to 1000, and that of force to 1. The starting learning rate is set to 0.0001, with a decay rate similar to the repo examples. I notice that both energy and force errors increase, without the energy error peak approaching zero.

**Reply by wanghan-iapcm** [reply]
I would suggest epref 1->1 or 1->10, force prefactor 1-> 1. 
How many steps did you use for the 1stage and 2stage trainings?
